utils/cacher.py

```diff
@@ -125,34 +145,44 @@ def create_cache_folder():
     if not os.path.exists('cache'):
         os.makedirs('cache')
 
-def write_project_data_to_file(project_ids: List[str], file_name: str):
+def write_project_data_to_file(project_keys: List[str], file_name: str):
     create_cache_folder()
 
     file = open("cache/" + file_name + ".txt", "a")
 
-    for project_id in project_ids:
-        file.write(project_id + "\n")
+    for project_key in project_keys:
+        file.write(project_key + "\n")
 
     file.close()
 
 def get_project_builds(cached_file: str) -> ProjectBuilds:
     file = open(cached_file, "r")
 
-    project_data = file.read().split("\n")
+    cached_projects = file.read().split("\n")
 
-    while("" in project_data):
-        project_data.remove("")
+    while("" in cached_projects):
+        cached_projects.remove("")
+
+    for i, project in enumerate(cached_projects):
+        cached_projects[i] = ProjectBuild(
+            display_name=project.split(",")[0],
+            key=project.split(",")[1],
+            project_type=project.split(",")[2],
+            is_valid_project=True,
+            org="",
+            state=""
+        )
 
-    for i, project in enumerate(project_data):
-        project_data[i] = ProjectBuild({
-            "name": project.split(",")[0],
-            "id": project.split(",")[1],
-            "type": project.split(",")[2],
-        })
+            # display_name: str
+            # key: str
+            # project_type: str
+            # is_valid_project: bool
+            # org: str
+            # state: str
 
     file.close()
```

Comment:
Consider refactoring to `with open(cached_file, "r") as file:` so that `close` is automatically called even if an exception is thrown.